Goder.BL/Services/ProblemService.cs

```diff
@@ -0,0 +1,72 @@
+using System;
+using System.Collections.Generic;
+using System.Text;
+using Goder.DAL.Context;
+using Goder.DAL.Models;
+using AutoMapper;
+using System.Linq;
+using Microsoft.EntityFrameworkCore;
+using System.Threading.Tasks;
+using Goder.BL.DTO;
+using Goder.BL.Exceptions;
+
+namespace Goder.BL.Services
+{
+    public class ProblemService : Abstract.BaseService
+    {
+        public ProblemService(GoderContext context, IMapper mapper) : base(context, mapper)
+        {
+
+        }
+        
+        public async Task<ProblemDTO> GetProblem(Guid id)
+        {
+            Problem problem = await _context.Problems.FirstOrDefaultAsync(c => c.Id == id);
```

Comment:
Here and in every place where you get problem you should use `Include` for getting user, tests, solutions (also you should send only few tests, not all, and send solutions only of specific user, if it is registered request)